cmd/aro/update_role_sets.go

```diff
@@ -0,0 +1,191 @@
+func getPlatformWorkloadIdentityRoleSet() (*api.PlatformWorkloadIdentityRoleSet, error) {
+	const envKey = envPlatformWorkloadIdentityRoleSets
+	var PlatformWorkloadIdentityRoleSet api.PlatformWorkloadIdentityRoleSet
+
+	// marshall env data into type api.PlatformWorkloadIdentityRoleSet
+	if err := getEnvironmentData(envKey, &PlatformWorkloadIdentityRoleSet); err != nil {
+		return nil, err
+	}
+
+	return &PlatformWorkloadIdentityRoleSet, nil
+}
+
+func getRoleSetFromEnv() ([]api.PlatformWorkloadIdentityRoleSet, error) {
+	roleSet, err := getPlatformWorkloadIdentityRoleSet()
+	if err != nil {
+		return nil, err
+	}
+
+	finalRoleSet := []api.PlatformWorkloadIdentityRoleSet{}
+	finalRoleSet = append(finalRoleSet, *roleSet)
+
+	return finalRoleSet, nil
+}
+
+// 2 - Get the existing role set documents, if existing
+// Mostly copied from update_ocp_versions.go
+func getPlatformWorkloadIdentityRoleSetDatabase(ctx context.Context, log *logrus.Entry) (database.PlatformWorkloadIdentityRoleSets, error) {
+	_env, err := env.NewCore(ctx, log, env.COMPONENT_UPDATE_OCP_VERSIONS)
+	if err != nil {
+		return nil, err
+	}
+
+	msiToken, err := _env.NewMSITokenCredential()
+	if err != nil {
+		return nil, fmt.Errorf("MSI Authorizer failed with: %s", err.Error())
+	}
+
+	msiKVAuthorizer, err := _env.NewMSIAuthorizer(_env.Environment().KeyVaultScope)
+	if err != nil {
+		return nil, fmt.Errorf("MSI KeyVault Authorizer failed with: %s", err.Error())
+	}
+
+	m := statsd.New(ctx, log.WithField("component", "update-role-sets"), _env, os.Getenv("MDM_ACCOUNT"), os.Getenv("MDM_NAMESPACE"), os.Getenv("MDM_STATSD_SOCKET"))
+
+	keyVaultPrefix := os.Getenv(envKeyVaultPrefix)
+	serviceKeyvaultURI := keyvault.URI(_env, env.ServiceKeyvaultSuffix, keyVaultPrefix)
+	serviceKeyvault := keyvault.NewManager(msiKVAuthorizer, serviceKeyvaultURI)
+
+	aead, err := encryption.NewMulti(ctx, serviceKeyvault, env.EncryptionSecretV2Name, env.EncryptionSecretName)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := env.ValidateVars(envDatabaseAccountName); err != nil {
+		return nil, err
+	}
+	dbAccountName := os.Getenv(envDatabaseAccountName)
+	clientOptions := &policy.ClientOptions{
+		ClientOptions: _env.Environment().ManagedIdentityCredentialOptions().ClientOptions,
+	}
+
+	logrusEntry := log.WithField("component", "database")
+	dbAuthorizer, err := database.NewMasterKeyAuthorizer(ctx, logrusEntry, msiToken, clientOptions, _env.SubscriptionID(), _env.ResourceGroup(), dbAccountName)
+	if err != nil {
+		return nil, err
+	}
+
+	dbc, err := database.NewDatabaseClient(log.WithField("component", "database"), _env, dbAuthorizer, m, aead, dbAccountName)
+	if err != nil {
+		return nil, err
+	}
+
+	dbName, err := DBName(_env.IsLocalDevelopmentMode())
+	if err != nil {
+		return nil, err
+	}
+	dbPlatformWorkloadIdentityRoleSetsDocument, err := database.NewPlatformWorkloadIdentityRoleSets(ctx, dbc, dbName)
```

Comment:
Nit: this is a client and not a document, so we could remove `Document` from the name for clarity.